sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.orc
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.catalog.HiveTableRelation
+import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRelation}
+import org.apache.spark.sql.execution.datasources.orc.OrcQueryTest
+import org.apache.spark.sql.hive.HiveUtils
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+import org.apache.spark.sql.internal.SQLConf
+
+class HiveOrcQuerySuite extends OrcQueryTest with TestHiveSingleton {
+  import testImplicits._
+
+  override def orcImp: String = "hive"
+
+  test("SPARK-8501: Avoids discovery schema from empty ORC files") {
```

Comment:
why does this test not in native orc test?

Reply:
Native ORC solve this bug. Native ORC have a corresponding test case [here](https://github.com/apache/spark/pull/19882/files#diff-9ab56940ee5a53f2bb81e3c008653362R554).
```
+  test("Schema discovery on empty ORC files") {
+    // SPARK-8501 is fixed.
```

Reply:
then why don't we put this test in the base class?

Reply:
This only works in new OrcFileFormat.
- The new test case is in `OrcQuerySuite` for new OrcFileFormat.
- And, the old test case is `HiveOrcQuerySuite` for old OrcFileFormat.